BDA696 Final/final.py

```diff
@@ -0,0 +1,1245 @@
+        frequency_predicted_svm,
+        bar_width,
+        alpha=opacity,
+        color="red",
+        label="Kernel SVM - Predicted",
+    )
+
+    rects6 = plt.bar(  # noqa
+        index + bar_width * 5,
+        frequency_predicted_xg,
+        bar_width,
+        alpha=opacity,
+        color="purple",
+        label="XGBoost - Predicted",
+    )
+
+    plt.xlabel("Result")
+    plt.ylabel("Frequency")
+    plt.title("Actual vs Predicted frequency.")
+    plt.xticks(index + bar_width, ("0", "1"))
+    plt.legend()
+
+    plt.tight_layout()
+    plt.savefig("/results/Comparative-of-different-classification-algorithms.png")
+    # plt.show()
+
+    # Feature importance
+
+    # RandomForestRegressor
+    model_RFR = RandomForestClassifier()
+    importance_RFR = model_RFR.fit(X, y).feature_importances_
+
+    feature_RFR = []
+    score_RFR = []
+    for i, v in enumerate(importance_RFR):
+        feature_RFR.append(f[i])
+        score_RFR.append(round(v, 5))
+
+    plot_feature_importance(importance_RFR, f, "RANDOM FOREST")
+
+    # DecisionTreeRegressor
+    model_DTR = DecisionTreeClassifier()
+    importance_DTR = model_DTR.fit(X, y).feature_importances_
+
+    feature_DTR = []
+    score_DTR = []
+    for i, v in enumerate(importance_DTR):
+        feature_DTR.append(f[i])
+        score_DTR.append(round(v, 5))
+    plot_feature_importance(importance_DTR, f, "DecisionTreeRegressor")
+
+    # XGBClassifier
+    model_XGB = XGBClassifier()
+    importance_XBG = model_XGB.fit(X, y).feature_importances_
+
+    feature_XGB = []
+    score_XGB = []
+    for i, v in enumerate(importance_XBG):
+        feature_XGB.append(f[i])
+        score_XGB.append(round(v, 5))
+    plot_feature_importance(importance_XBG, f, " XBGoost")
+
+    # Variance Inflation Factor (VIF)
+
+    df_vif = df_M.drop(columns=["home_team_wins"])
+
+    VIF = [vif(df_vif.values, i) for i in range(len(df_vif.columns))]
+
+    feature = []
+    score = []
+    for i, v in enumerate(VIF):
+        feature.append(f[i])
+        score.append(round(v, 1))
+
+    pd.DataFrame(score, columns=["VIF"], index=df_vif.columns).sort_values(
+        by=["VIF"], ascending=False
+    )
+
+    plot_vif(VIF, df_vif.columns, "VIF")
+
+    percentageM = []
+    for m in MeanSquaredDiffWeighted:
+        percentageM.append(round(m * 100, 3))
+    percentageM
+    df_ranking_vartype = pd.DataFrame({"Feature": pd.Series(feature)})
+    df_ranking_vartype = df_ranking_vartype.assign(
+        predictor_type=pd.Series(predictor_type),
+        Score_DTR=pd.Series(score_DTR),
+        score_RFR=pd.Series(score_RFR),
+        Score_XGB=pd.Series(score_XGB),
+        t_score=pd.Series(t_score),
+        VIF=pd.Series(VIF),
+        MSDw=pd.Series(percentageM),
+    )
+
+    df_ranking_vartype.sort_values(
+        by=["MSDw", "t_score"], ascending=False
+    ).to_csv(  
+        r"/results/baseball.csv"
+    )
+
+    print(df_ranking_vartype)
+
+
+if __name__ == "__main__":
+    sys.exit(main())

```

Comment:
It looks like you wrote this in jupyter notebook or something like that before committing it. It runs fine, but there's a lot of single lines that look like they were used to print something.